### PR TITLE
refactor: [M3-8216] - NodeBalancer Query Key Factory

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Updates NodeBalancer queries to use a Query Key Factory and latest TanStack query patterns

## Preview 📷

> [!note]
> No UI changes expected

## How to test 🧪

- Generally test all NodeBalancer related features
- Test assigning / unassigning NodeBalancers from Firewalls 
- Verify all NodeBalancer automated tests pass

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by cpathipa — APPROVED** [review]
Overall LGTM! 

- Verified NodeBalancer functionality 
- Verified assigning / unassigning NodeBalancers from Firewalls

File: packages/manager/src/queries/nodebalancers.ts [hunk 1]
```diff
@@ -25,126 +19,178 @@ import {
...
+  return useMutation<NodeBalancer, APIError[], CreateNodeBalancerPayload>({
+    mutationFn: createNodeBalancer,
+    onSuccess(nodebalanacer) {
+      // Invalidate paginated stores
+      queryClient.invalidateQueries({
+        queryKey: nodebalancerQueries.nodebalancers.queryKey,
+      });
+      // Prime the cache for this specific NodeBalancer
+      queryClient.setQueryData<NodeBalancer>(
+        nodebalancerQueries.nodebalancer(nodebalanacer.id).queryKey,
```
Comment (mjac0bs) [inline packages/manager/src/queries/nodebalancers.ts:168]:
```suggestion
        nodebalancerQueries.nodebalancer(nodebalancer.id).queryKey,
```

File: packages/manager/src/queries/nodebalancers.ts
Comment (mjac0bs) [inline packages/manager/src/queries/nodebalancers.ts:1]: We should probably just find-and-replace to fix that typo that has bugged me for a while. 😄  🧹

**Review by mjac0bs — COMMENTED** [review]
Widespread NodeBalancer PRs always make me a little nervous because our e2e coverage is lacking for the feature. We really only test the basics of the create flow and empty state landing. It would be really nice to have test coverage for the landing page and details pages, but I think those exist somewhere in the backlog. 

Spent some time testing functionality, including NBFW CRUD interactions, and things were looking good for the most part. See my comment about the issue with the `getAllNodeBalancerConfigs` query that was causing regressions.

**Comment by bnussman-akamai** [comment]
I think I got all of that addressed! Thanks for catching those ✏️  @mjac0bs

**Review by mjac0bs — APPROVED** [review]
Thanks @bnussman-akamai, all looks good now! 🚢

**Pull request merged**